iree/compiler/Dialect/HAL/Target/VMLA/VMLATarget.cpp

```diff
@@ -66,6 +90,191 @@ class VMLATargetBackend final : public TargetBackend {
         passManager, IREE::VM::getTargetOptionsFromFlags());
   }
 
+  LogicalResult linkExecutables(mlir::ModuleOp moduleOp) override {
+    // --- Linking overview ---
+    //
+    // We start with a `module` containing multiple `hal.executable`s, each with
+    // potentially multiple `hal.executable.target`s. We want to move all
+    // compatible VMLA functions into a new "linked" executable, de-duping
+    // symbols, and updating references as we go.
+    //
+    // Sample IR before:
+    //   hal.executable @main_dispatch_0 {
+    //     hal.interface @legacy_io { ... }
+    //     hal.executable.target @vmla, filter="vmla" {
+    //       hal.executable.entry_point @main_dispatch_0 attributes { ... }
+    //       module { vm.module @module { vm.func @main_0(...) { ... } } }
+    //     }
+    //     hal.executable.target @other, filter="other" {
+    //       hal.executable.entry_point @main_dispatch_0 attributes { ... }
+    //       module { ... }
+    //     }
+    //   }
+    //
+    // Sample IR after:
+    //   hal.executable @linked_vmla {
+    //     hal.interface @legacy_io { ... }
+    //     hal.executable.target @vmla, filter="vmla" {
+    //       hal.executable.entry_point @main_dispatch_0 attributes { ... }
+    //       hal.executable.entry_point @main_dispatch_1 attributes { ... }
+    //       hal.executable.entry_point @main_dispatch_2 attributes { ... }
+    //       module {
+    //         vm.module @module {
+    //           vm.func @main_0(...) { ... }
+    //           vm.func @main_1(...) { ... }
+    //           vm.func @main_2(...) { ... }
+    //         }
+    //       }
+    //     }
+    //   }
+    //   hal.executable @main_dispatch_0 {
+    //     hal.interface @legacy_io { ... }
+    //     hal.executable.target @other, filter="other" {
+    //       hal.executable.entry_point @main_dispatch_0 attributes { ... }
+    //       module { ... }
+    //     }
+    //   }
```

Comment:
I think I'll write a lit test that shows this [instead?]

Reply:
yeah the advantage of lit is that it's always up to date but can be so unreadable with all the stupid capture syntax that it loses its utility as a comment. you could truncate this a bit and only show the final form here and then have the test cover the specific cases you want to highlight.

Reply:
Added a lit test specifically for VMLA linking. Still need to revise this comment and fix other newly failing tests.